src/Azure.Functions.Cli/Actions/LocalActions/CreateProxyAction.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Functions.Cli.Interfaces;
+using Azure.Functions.Cli.Common;
+using Colors.Net;
+using Fclp;
+using static Azure.Functions.Cli.Common.OutputTheme;
+using System.Net.Http;
+using System.IO;
+using System.Reflection;
+
+namespace Azure.Functions.Cli.Actions.LocalActions
+{
+    [Action(Name = "create", Context = Context.Proxy, HelpText = "Create a new function proxy.")]
+    [Action(Name = "new", Context = Context.Proxy, HelpText = "Create a new function proxy.")]
+    internal class CreateProxyAction : BaseAction
+    {
+        private readonly IProxyManager _proxyManager;
+
+        public string ProxyName { get; set; }
+        public List<string> Methods { get; set; }
+        public string Route { get; set; }
+        public Uri BackendUrl { get; set; }
+        public string TemplateName { get; set; }
+
+        public CreateProxyAction(IProxyManager proxyManager)
+        {
+            _proxyManager = proxyManager;
+        }
+
+        public override ICommandLineParserResult ParseArgs(string[] args)
+        {
+            Parser
+                .Setup<string>('n', "name")
+                .WithDescription("Proxy name")
+                .Callback(n => ProxyName = n);
+            Parser
+                .Setup<string>('r', "route")
+                .WithDescription("Route template")
+                .Callback(r => Route = r);
+            Parser
+                .Setup<List<string>>("methods")
+                .WithDescription("One or more http methods(space delimited).")
+                .Callback(m => Methods = m);            
+            Parser
+                .Setup<Uri>("backend-url")
```

Comment:
for sake of consistency with other actions in the CLI, make these options camelCase